gammapy/irf/edisp_map.py

```diff
@@ -345,24 +255,176 @@ def from_diagonal_response(cls, energy_axis_true, migra_axis=None):
 
         return cls.from_geom(geom)
 
-    def cutout(self, position, width, mode="trim"):
-        """Cutout edisp map.
+    def to_edisp_kernel_map(self, energy_axis):
+        """Convert to map with edisp kernels
+
+        Parameters
+        ----------
+        e_reco : `MapAxis`
+            Reconstructed enrgy axis.
+
+        Returns
+        -------
+        edisp : `EDispKernelMap`
+            Energy dispersion kernel map.
+        """
+        axis = 0
+        energy_axis_true = self.edisp_map.geom.get_axis_by_name("energy_true")
+        migra_axis = self.edisp_map.geom.get_axis_by_name("migra")
+
+        data = []
+
+        for idx, e_true in enumerate(energy_axis_true.center):
+            # migration value of e_reco bounds
+            migra = energy_axis.edges / e_true
+
+            edisp_e_true = self.edisp_map.slice_by_idx({"energy_true": idx})
+
+            cumsum = np.insert(edisp_e_true.data, 0, 0, axis=axis).cumsum(axis=axis)
+            with np.errstate(invalid="ignore"):
+                cumsum = np.nan_to_num(cumsum / cumsum[slice(-2, -1)])
+
+            f = interp1d(
+                migra_axis.edges.value,
+                cumsum,
+                kind="linear",
+                bounds_error=False,
+                fill_value=(0, 1),
+                axis=axis
+            )
+
+            integral = np.diff(np.clip(f(migra), a_min=0, a_max=1), axis=axis)
+            data.append(integral)
+
+        data = np.stack(data)
+
+        geom_image = self.edisp_map.geom.to_image()
+        geom = geom_image.to_cube([energy_axis, energy_axis_true])
+        edisp_kernel_map = Map.from_geom(geom=geom, data=data)
+        return EDispKernelMap(
+            edisp_kernel_map=edisp_kernel_map, exposure_map=self.exposure_map
+        )
+
+
+class EDispKernelMap(IRFMap):
+    """Energy dispersion kernel map.
+
+    Parameters
+    ----------
+    edisp_kernel_map : `~gammapy.maps.Map`
+        The input energy dispersion kernel map. Should be a Map with 2 non spatial axes.
+        Reconstructed and and true energy axes should be given in this specific order.
+    exposure_map : `~gammapy.maps.Map`, optional
+        Associated exposure map. Needs to have a consistent map geometry.
+
+    """
+    _hdu_name = "edisp"
+
+    def __init__(self, edisp_kernel_map, exposure_map):
+        if edisp_kernel_map.geom.axes[1].name.upper() != "ENERGY_TRUE":
+            raise ValueError("Incorrect energy axis position in input Map")
+
+        if edisp_kernel_map.geom.axes[0].name.upper() != "ENERGY":
+            raise ValueError("Incorrect migra axis position in input Map")
+
+        super().__init__(irf_map=edisp_kernel_map, exposure_map=exposure_map)
+
+    @property
+    def edisp_map(self):
+        return self._irf_map
+
+    @edisp_map.setter
+    def edisp_map(self, value):
+        self._irf_map = value
+
+    @classmethod
+    def from_geom(cls, geom):
```

Comment:
Again this was introduced for homogeneity with the existing `EDispMap`.  At least it's documented that a diagonal energy response is created by default...

Do you think there is another reasonable default instead of diagonal? E.g. filling with zeros? I think only this case a differentiation of the names would be needed...